subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,85 @@
+ project('speexdsp', 'c',
+  version : '1.2.0',
+  default_options: ['warning_level=2',
+                    'c_std=gnu99',
+                    'buildtype=release'],
+  license : 'bsd-like')
+
+lib_current = 1
+lib_revision = 0
+lib_age = 2
```

Comment:
I am not sure of the versioning scheme used by this project, but it seem that this could be
```
v = meson.project_version().split('.')
lib_current = v[0]
lib_age = v[1]
lib_revision = v[2]
```
avoiding to encode the project version in two places. But `lib_current`, `lib_age`, and `lib_revision` seem to be used only to define `lib_version`, thus see below.

Reply:
The versioning scheme used by this project is autotools, so prepare for great confusion and annoyance.

The lib_current / lib_revision / lib_age is wrong as defined here. Upstream calculates this as `6, 1, 5`, and passes it to libtool as `-version-info 6:1:5`.

libtool then does math (specifically, `(C - A).(A).(R)`), and sets the equivalent of Meson's `version: '1.5.1'`, which you'll notice is not the same as either the libtool numbers (because math!!!) or the project version.

Given the project version and the library version are totally unconnected, you MUST encode it in two places, the only question is whether to do it via math or as a string constant.

Reply:
I started this wrap based on the `ogg` files, so that's where this `lib_` stuff came from -- I'm removing all of it here because it's confusing and not adding any value that I can see. Thanks for the explanation @eli-schwartz !

Reply:
The ogg wrap does the math in question, and rationalizes using the confusion because... it allows trivially copying the values of C, A, and R from the configure.ac on updates. This may or may not be more or less confusing than checking the final filename of an autotools build, and then copying that as a statically coded string.

You can't remove *all* of it, you do need to pass the `version: ` kwarg somehow, whether you statically code it as `'1.5.1'` or copy in the C, A, R values from configure.ac.

For reference, the ogg files aren't *that* confusing. Here's what they do:

```meson
# See configure.ac and CMakeLists.txt
lib_current = 8
lib_revision = 5
lib_age = 8
lib_soversion = lib_current - lib_age
lib_version = '@0@.@1@.@2@'.format(lib_soversion, lib_age, lib_revision)
```

and the first 3 variables can be copy/pasted from configure.ac to suit.

Reply:
thanks for going into more detail, @eli-schwartz .

Going with the C, A, and R copy-and-paste approach [from the speexdsp-1.2.0's configure.ac](https://github.com/xiph/speexdsp/blob/68311d46785be76d2a186c75578d51108bff6dfb/configure.ac#L3-L14):

```
lib_current = 6   
lib_revision = 1
lib_age = 5
lib_soversion = lib_current - lib_age
lib_version = '@0@.@1@.@2@'.format(lib_soversion, lib_age, lib_revision)
```

The math escapes me:

 - `lib_soversion` = 6 - 5 => **1**
 - `lib_age` = **5**
 - `lib_revision` = **1**

Putting this together into `lib_version =` gives me `1.5.1`, which I suppose matches the major version  of SpeexDSP's 1.2.0 -- but I've been even more confused by other autotool'isms than this :sweat_smile: